clients/stellar-relay-lib/src/connection/connector/message_handler.rs

```diff
@@ -24,7 +23,7 @@ impl Connector {
 		match msg_type {
 			MessageType::Transaction | MessageType::FloodAdvert if !self.receive_tx_messages() => {
 				self.increment_remote_sequence()?;
-				self.check_to_send_more(MessageType::Transaction).await?;
+				self.check_to_send_more(MessageType::Transaction, data.len()).await?;
```

Comment:
Unlike how the data length of the message is calculated [here](https://github.com/stellarbeat/js-stellar-node-connector/blob/master/src/connection/connection.ts#L316), I believe in our implementation we already have the "cleaned" `XDR` representation at this point, and does not need to be stripped like it is done [here](https://github.com/stellarbeat/js-stellar-node-connector/blob/master/src/connection/connection.ts#L380).

Reply:
Let's see. What I can tell is that I tried before to substract the "padding" (as is calculated in the js implementation) and we get a panic since the data length is smaller. So if the message itself is padded, it is not exactly by those amount of bytes.

Reply:
I think I was wrong, please check the last committed [change](https://github.com/pendulum-chain/spacewalk/pull/550/commits/7a4a105c8f3922984aca6a711de77dbfd49cdafc), now it should be good. 

Two main things that hinted this:
- We do the same [slicing](https://github.com/pendulum-chain/spacewalk/blob/d74009bf1d8c27fbd1f314c4fe486cd230e8b73f/clients/stellar-relay-lib/src/connection/connector/message_handler.rs#L51) to verify [authentication](https://github.com/stellarbeat/js-stellar-node-connector/blob/9120f24b75867700ba0ece76369166959205c6b9/src/connection/connection.ts#L372)
- We slice also before parsing the message ([ours](https://github.com/pendulum-chain/spacewalk/blob/main/clients/stellar-relay-lib/src/connection/xdr_converter.rs#L64), [theirs](https://github.com/stellarbeat/js-stellar-node-connector/blob/9120f24b75867700ba0ece76369166959205c6b9/src/connection/connection.ts#L380)).

The reason it was failing, is because the call to `check_send_more` [here](https://github.com/pendulum-chain/spacewalk/pull/550/commits/7a4a105c8f3922984aca6a711de77dbfd49cdafc#diff-a415e544a5cbc568c7c5b061585fcbe1650f2edc4dabaf6548e3e7eeb67cb13aL98) was incorrect. Also, I added these two [cases](https://github.com/pendulum-chain/spacewalk/pull/550/commits/7a4a105c8f3922984aca6a711de77dbfd49cdafc#diff-a415e544a5cbc568c7c5b061585fcbe1650f2edc4dabaf6548e3e7eeb67cb13aR96) on the match (as the [original](https://github.com/stellarbeat/js-stellar-node-connector/blob/9120f24b75867700ba0ece76369166959205c6b9/src/connection/connection.ts#L509)), even though we do nothing with them, it would not be correct to call the control on them.

Reply:
True, nice catches 👌 makes sense to me